Docker env variable

Berserkir-Wolf opened the pull request:
To resolve #33 (and #35 as a side-effect).

System will generate a user with sudo permissions inside the container, but with whatever UID and GID you define as environment variables (docs modification to follow later). Default UID/GID inside the container is 1000 (with this modification).

Note that this change does not break existing deployments providing the UID has permission to the directories in place (UID 1000 is the first non-root user created on a debian-based system by default).

Casvt commented:
A few things:

1. Have you tested the PUID:PGID changes? Does it work?
2. In the docs there is an extra tab for if you want to specify PUID:PGID in the docker-compose file, but no instructions on how to do this for the Docker CLI command. 
3. All the changes you suggested in your other docs-PR are also made here, why?
4. The `<username>` found starting from [here](https://github.com/Casvt/Kapowarr/pull/55/files#diff-8b042e3f94ca5c59a7cd990b950aec0073ea84fe811e7b22be51158d7b180d56R156) don't display in the resulting docs. Install the `docs-requirements.txt` file and run `mkdocs serve` in a terminal to test your changes locally.
5. Remove the unzipping block

Casvt commented on docs/installation.md:
```diff
@@ -46,6 +46,27 @@ The recommended way to install Kapowarr is using Docker. After installing Kapowa
+    docker run -d \
+        --name kapowarr \
+        -e PUID:1000 \
+        - GUID:1000 \
```
Should be `-e`

Berserkir-Wolf replied:
I was partway through writing the amendment to that one already.

Berserkir-Wolf commented:
1. I have, it does.
2. Added in c24502b
3. I didn't switch back to the dev branch when creating this one - boss came out and distracted me at lunch.
4. Fixed in 09dca35
5. Gone in 3740aaf
6. If you set PUID/PGID this way, the application runs under a user with that ID inside the container. 
When it then saves/adjusts files, they get placed in the mounted volume as being attached to that same UID/GID, which has a different mapping in the host system.  
I have mediauser:mediagroup setup in my test setup, and when setting the IDs to match then the files show as being owned by the aforementioned group. It's a quick fix, in preparation for the new `-username` option that docker now has.
I say 'new' only because there are still relatively few references on how to switch a container to use it rather than the old PUID:PGID method. Not even linuxserver have switched over yet. As I slowly get my head around that one, I'll be modifying the dockerfile in my test environment to see if I can make it work there.

Berserkir-Wolf commented:
18146b0 fixed the 'building a library' conflict - only a merge because it was done in the web editor rather than VS Code.

ggrensteiner commented:
If this has been merged it still doesn't work. PGID and PUID are ignored in docker-compose and the closest thing to docs about it give the wrong format for docker-compose

```
    environment:
     - PUID: 1000
     - PGID: 1000

```
is not valid syntax

```
    environment:
     - PUID=1000
     - PGID=1000
```
is ignored and the container runs as root and writes files as root.

Casvt commented:
These changes have been removed for a long time because they indeed don't work. The current new PR #199 for this feature also doesn't actually work. Support for it is planned for V1.3.0.
